src/lib/Handshake.js

```diff
@@ -311,11 +320,7 @@ class Handshake {
 
     this._handshakeStage = 'send-session-key';
```

Comment:
I think we might be able to get rid of `_handshakeStage` as well. I don't think we really use it anymore.